src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -800,8 +800,13 @@ def subvolume_snapshot_getpath(self, **kwargs):
             with open_subvol_in_vol(self, self.volspec, volname, groupname, subvolname,
                                     SubvolumeOpType.SNAP_GETPATH) \
                                     as (vol, group, subvol):
-                snap_path = subvol.snapshot_data_path(snapname)
-                ret = 0, snap_path.decode("utf-8"), ""
+                if subvol.version() == 1:
+                    ret = (-errno.ENOTSUP, '',
+                           'command not supported for v1 subvolumes')
+                    return ret
```

Comment:
Normally I would expect v1 to implement `snapshot_data_path` as a method which always returns `-ENOTSUP` rather than special case here.

Won't stop approval though. Please clean up in a future PR.

Reply:
Also, let's add another TODO to have some command from the volumes plugin tell us how many v1/v2/v3 subvols there (any upgrades done, if applicable). That could be collected by telemetry.

Reply:
> Normally I would expect v1 to implement snapshot_data_path as a method which always returns -ENOTSUP rather than special case here.

snapshot_data_path() is already implemented which is why I added this block of code here  - https://github.com/ceph/ceph/blob/main/src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py#L76-L78

Reply:
Wait I'm confused now. You're saying v1 **could** support the command but we're choosing not to?

https://github.com/ceph/ceph/blob/main/src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py#L84-L86

Reply:
I think the concern was something related to retained snaps in v1.